aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/s3/S3ExpressDecorator.kt

```diff
@@ -198,6 +212,67 @@ class S3ExpressIdentityProviderConfig(codegenContext: ClientCodegenContext) : Co
         }
 }
 
+class S3ExpressRequestChecksumCustomization(
+    private val codegenContext: ClientCodegenContext,
+    private val operationShape: OperationShape,
+) : OperationCustomization() {
+    private val runtimeConfig = codegenContext.runtimeConfig
+    private val inputShape = codegenContext.model.expectShape(operationShape.inputShape)
+
+    private val codegenScope =
+        arrayOf(
+            *preludeScope,
+            "ChecksumAlgorithm" to RuntimeType.smithyChecksums(runtimeConfig).resolve("ChecksumAlgorithm"),
+            "ConfigBag" to RuntimeType.configBag(runtimeConfig),
+            "DefaultRequestChecksumOverride" to
+                runtimeConfig.awsInlineableHttpRequestChecksum()
+                    .resolve("DefaultRequestChecksumOverride"),
+            "Document" to RuntimeType.smithyTypes(runtimeConfig).resolve("Document"),
+        )
+
+    override fun section(section: OperationSection): Writable =
+        writable {
+            when (section) {
+                is OperationSection.AdditionalRuntimePluginConfig -> {
+                    rustTemplate(
+                        """
+                        ${section.newLayerName}.store_put(#{DefaultRequestChecksumOverride}::new(
+                            |original: #{Option}<#{ChecksumAlgorithm}>,
+                            cfg: &#{ConfigBag}| {
+                                if original != #{Some}(#{ChecksumAlgorithm}::Md5) {
+                                    return original;
+                                }
+
+                                let endpoint = cfg
+                                    .load::<crate::config::endpoint::Endpoint>()
+                                    .expect("endpoint added to config bag by endpoint orchestrator");
+
+                                match endpoint.properties().get("backend") {
+                                    Some(#{Document}::String(backend)) if backend.as_str() == "S3Express" => {
+                                       #{customDefault:W}
```

Comment:
If you decide to fork `RequestChecksumInterceptor` for S3, then you can just look up the request `Metadata` in the config bag to get the operation name. Or, alternatively, you can have the operation runtime plugin for that operation put the override value into the config bag.